packages/signature-controller/src/SignatureController.ts

```diff
@@ -880,4 +897,67 @@ export class SignatureController extends BaseController<
 
     return networkClient.configuration.chainId;
   }
+
+  #decodePermitSignatureRequest(
+    signatureRequestId: string,
+    request: OriginalRequest,
+    chainId: string,
+  ) {
+    // Code below will invoke signature request decoding api for permits
+    if (!this.#decodingApi) {
+      return;
+    }
+    let decodedRequest: DecodedRequestInfo;
+    try {
+      const { primaryType } = JSON.parse(request.params?.[1] ?? '');
+      if (primaryType === 'Permit') {
```

Comment:
Why are we limiting this to a specific primary type? Even if the API didn't support a type, should it not just return an empty response?

We don't want to maintain this here for every primary type it supports.

Reply:
I changed it to check for method `eth_signTypedData_v4`.